### PR TITLE
feat: add support for outgoing webhook custom http headers

### PR DESCRIPTION
**Pull request by JeevaRamu0104**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
image

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by gitanjli525 — APPROVED** [review]

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 1]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+  @react.component
+  let make = (~removeAuthHeaders, ~index) => {
+    let (key, setKey) = React.useState(_ => "")
+    let (metaValue, setValue) = React.useState(_ => "")
+    let form = ReactFinalForm.useForm()
+    let name = `outgoing_webhook_custom_http_headers.${key}`
+    let keyInput: ReactFinalForm.fieldRenderPropsInput = {
+      name: "string",
+      onBlur: _ev => {
+        if key->Js.String.length > 0 {
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:39]:
```suggestion
        if key->String.length > 0 {
```

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 2]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+module AuthenticationInput = {
+  @react.component
+  let make = (~removeAuthHeaders, ~index) => {
+    let (key, setKey) = React.useState(_ => "")
+    let (metaValue, setValue) = React.useState(_ => "")
+    let form = ReactFinalForm.useForm()
+    let name = `outgoing_webhook_custom_http_headers.${key}`
+    let keyInput: ReactFinalForm.fieldRenderPropsInput = {
+      name: "string",
+      onBlur: _ev => {
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:38]: If not using remove it

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 3]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+        if key->Js.String.length > 0 {
+          form.change(name, metaValue->JSON.Encode.string)
+        }
+      },
+      onChange: ev => {
+        let value = ReactEvent.Form.target(ev)["value"]
+        setKey(_ => value)
+      },
+      onFocus: _ev => {
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:47]: If not using remove it

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 4]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+        if key->Js.String.length > 0 {
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:48]:
```suggestion
        if key->String.length > 0 {
```

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 5]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+          form.change(name, JSON.Encode.null)
+        }
+      },
+      value: key->JSON.Encode.string,
+      checked: true,
+    }
+
+    let valueInput: ReactFinalForm.fieldRenderPropsInput = {
+      name: "string",
+      onBlur: _ev => {
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:58]:
```suggestion
      onBlur: _ => {
```

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 6]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+        form.change(name, metaValue->JSON.Encode.string)
+      },
+      onChange: ev => {
+        let value = ReactEvent.Form.target(ev)["value"]
+        setValue(_ => value)
+      },
+      onFocus: _ev => (),
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:65]:
```suggestion
      onFocus: _ => (),
```

File: src/screens/Settings/MerchantAccountUtils.res [hunk 7]
```diff
@@ -132,6 +132,20 @@ let getBusinessProfilePayload = (values: JSON.t) => {
     valuesDict->getString("three_ds_requestor_url", "")->getNonEmptyString,
   )
 
+  let outGoingWebHookCustomHttpHeaders = Dict.make()
+  let formValues = valuesDict->getDictfromDict("outgoing_webhook_custom_http_headers")
+
+  let _ =
```
Comment (PritishBudhiraja) [inline src/screens/Settings/MerchantAccountUtils.res:138]: It won't return anything.
So we can remove let _ =

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 8]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+      if authHeaders->Array.length < 4 {
+        let update = authHeaders->Array.concat([authHeaders->Array.length])
+        setAuthHeaders(_ => update)
+      }
+    }
+
+    <div className="flex-1">
+      {authHeaders
+      ->Array.mapWithIndex((_, index) => {
+        <div className="grid grid-cols-5 flex gap-4">
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:134]: Add key

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 9]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+            <AuthenticationInput removeAuthHeaders index />
+          </div>
+          <UIUtils.RenderIf condition={index === authHeaders->Array.length - 1 && index != 3}>
+            <div className="flex justify-start items-center mt-4">
+              <Icon
+                name="plus-circle"
+                size=27
+                customIconColor="text-gray-400"
+                className="flex items-center justify-center w-fit h-fit"
+                onClick={_ev => addAuthHeaders()}
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:145]: same

**Review by PritishBudhiraja — COMMENTED** [review]
Comments added

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 10]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+      <UIUtils.RenderIf condition={index > 0}>
+        <div className="mt-6 flex gap-4">
+          <ModalCloseIcon onClick={_ev => removeAuthHeaders(index, key)} />
+        </div>
+      </UIUtils.RenderIf>
+      <UIUtils.RenderIf condition={index == 0}>
+        <div className="flex bg-transparent text-white items-center mt-4">
+          <p> {"text"->React.string} </p>
+        </div>
+      </UIUtils.RenderIf>
```
Comment (Riddhiagrawal001) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:88]:
```suggestion
      <UIUtils.RenderIf condition={index == 0}>
       <div className="flex bg-transparent text-white items-center mt-4 h-4 w-8" />
      </UIUtils.RenderIf>
```

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 11]
```diff
@@ -26,6 +26,190 @@ module InfoViewForWebhooks = {
...
+      None
+    }, [formState.values])
+
+    let removeAuthHeaders = (removeIndex, name) => {
+      let outGoingWebhookDict =
+        formValuesRef.current
+        ->getDictFromJsonObject
+        ->getDictfromDict("outgoing_webhook_custom_http_headers")
+
+      let _ = outGoingWebhookDict->Dict.delete(name)
```
Comment (sagarnaikjuspay) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:115]: use `->ignore`

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 12]
```diff
@@ -143,18 +327,34 @@ let make = (~webhookOnly=false, ~showFormOnly=false, ~profileId="") => {
...
-                    <FormRenderer.FieldRenderer
-                      key={index->Int.toString}
-                      field
-                      errorClass
-                      labelClass="!text-base !text-grey-700 font-semibold"
-                      fieldWrapperClass="max-w-xl"
-                    />
-                  )
-                  ->React.array}
+                  <FormRenderer.FieldRenderer
```
Comment (sagarnaikjuspay) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:330]: `open FormRenderer` at top

File: src/screens/Settings/MerchantAccountUtils.res (hunk 7 above)
Comment (sagarnaikjuspay) [reply src/screens/Settings/MerchantAccountUtils.res:138]: It should've showned warning right

Comment (JeevaRamu0104) [reply src/screens/Settings/MerchantAccountUtils.res:145]: yes

**Review by gitanjli525 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**